arviz/data/inference_data.py

```diff
@@ -177,6 +177,11 @@ def __delattr__(self, group):
     def _groups_all(self):
         return self._groups + self._groups_warmup
 
+    def __iter__(self):
+        for group in self._groups_all:  # pylint: disable=protected-access
```

Comment:
Are these still needed? `.pylintrc` should have a ignore for these.

Reply:
Yeah, it ignores all these. I have added it as a habit. WIll remove.